File: Java/MyFirstConnection.java
import xrtml.Config;
import xrtml.Console;
import xrtml.channel.Channel;
import xrtml.channel.events.OnMessageEvent;
import xrtml.channel.events.OnSubscribeEvent;
import xrtml.channel.interfaces.IOnMessage;
import xrtml.channel.interfaces.IOnSubscribe;
import xrtml.connection.Connection;
import xrtml.connection.events.OnConnectEvent;
import xrtml.connection.interfaces.IOnConnect;
import xrtml.connectionmanager.ConnectionManager;

public class MyFirstConnection {

	public static void main(String[] args) {

		// This example shows how to create a connection, bind event listeners to it and send messages.

		// To start we active the debug to see some useful logs.
		Config.setDebug(true);

		// Then we get the ConnectionManager singleton.
		ConnectionManager connectionManager = ConnectionManager.getInstance();

		// Now we create a connection.
		final Connection myConnection = connectionManager.create("myConnection", "myAppKey", "myAuthToken", "http://ortc-developers.realtime.co/server/2.1");

		// Here we define a listener for the connect event.
		myConnection.bind(new IOnConnect() {
			@Override
			public void run(OnConnectEvent evt) {

				// In the connect event we want to subscribe the channel "myChannel"
				((Connection) evt.getSource()).subscribe("myChannel");
			}
		});

		// Here we create a channel.
		final Channel myChannel = new Channel("myChannel");
		// We set the subscribe flag to false so it does not subscribe automatically.
		myChannel.setSubscribe(false);

		// Here we define a listener to the subscribe channel event and a listener to the message channel event
		myChannel.bind(new IOnSubscribe() {
			@Override
			public void run(OnSubscribeEvent evt) {
				myConnection.send("myChannel", "MyTestMessage");
			}
		});

		myChannel.bind(new IOnMessage() {
			@Override
			public void run(OnMessageEvent evt) {
				Console.info(evt.getSource().getClass(), evt.getMessage());
			}
		});

		// Then we associate the channel to the connection.
		myConnection.createChannel(myChannel);
		// And finally we connect.
		myConnection.connect();
	}
}

File: NodeJS/MyFirstConnection.js
// In this example we show how to create a connection, bind event listeners to the connection and send messages.

// First we need to require the xRTML module.
// Install it via Node Package Manager and then require it
var xRTML = require("xRTML");

// One of the modules that xRTML contains is the ConnectionManager. The Connection manager module contains the functions to create and manage connections.
// So we are using it to create a simple connection.
var myConnection = xRTML.ConnectionManager.create({
	// Here we specify the url of the ORTC server.
    url: "http://ortc-developers.realtime.co/server/2.1",
    // The id of the connection.
    id: "myConnection",
    // The application key 
    appKey: 'myAppKey',
    // The authentication token
    authToken: 'myAuthToken',
    // The list of channels assigned to the connection.
    channels: [{name: 'myChannel', subscribe: false }]
});

// Every connection hold the event mechanism, so we can bind/unbind event listeners to them.
// Here is a simple example that shows how to do it.
myConnection.bind({
	// type and the listener function
	connect: function(e) {
		// At this point the connection is established. Every connection must contain at least a channel subscribed to receive messages.
		// As you may have realized, when we  we set the subscribe value of the channel "myChannel" intentionally to false.
		// Now we are going to bind some event listeners and subscribe it.

		// We will bind to the event subscribe and to the event message.
		this.channels.myChannel.bind({			
			subscribe: function(e){

				console.log("Channel [" + this.name +"] is subscribed.");

				// Now that we have a subscribed channel, we are going to send a message to that channel.
				myConnection.send({
					channel: "myChannel",
					content: "MyTestMessage"
				});
			},
			message: function(e){
				// At this point, a message has arrived to the channel.
				console.log("A message has arrived to the channel [" + this.name + "]");
				console.log(e.message);
			}
		});

		this.subscribe({name: "myChannel"});
	}
});
// And thats it. We covered the basic connection steps.